importers/research-importer.php

```diff
@@ -0,0 +1,233 @@
+<?php
+/**
+ * Research and Researcher importer
+ */
+namespace UCF\MainSiteUtilities\Importers {
+	class ResearchImporter {
+		private
+			$api_url,
+			$params,
+			$researchers,
+
+			// Researcher counts
+			$researchers_processed = 0,
+			$researchers_created   = 0,
+			$researchers_updated   = 0,
+			$researchers_skipped   = 0;
+
+		/**
+		 * Constructor
+		 * @author Jim Barnes
+		 * @since 2.0.0
+		 * @param string $search_service_url The base url of the search service
+		 * @param array $params Additional parameters to pass to the search service
+		 */
+		public function __construct( $search_service_url, $params ) {
+			$this->api_url = \trailingslashit( $search_service_url );
+			$this->params = $params;
+			$this->researchers = array();
+		}
+
+		/**
+		 * The import function
+		 * @author Jim Barnes
+		 * @since 2.0.0
+		 */
+		public function import() {
+			$this->get_researchers();
+			$this->import_researchers();
+		}
+
+		/**
+		 * Returns the formatted string of
+		 * stats from the import
+		 * @author Jim Barnes
+		 * @since 2.0.0
+		 * @return string
+		 */
+		public function get_stats() {
+			return "
+Researchers
+-----------
+
+Processed: {$this->researchers_processed}
+Created  : {$this->researchers_created}
+Updated  : {$this->researchers_updated}
+Skipped  : {$this->researchers_skipped}
+
+			";
+		}
+
+		/**
+		 * Helper function that retrieves JSON
+		 * results from a URL
+		 * @author Jim Barnes
+		 * @since 2.0.0
+		 * @param string $url The URL to fetch the results from
+		 * @param mixed Returns an object if parsed, false if failed
+		 */
+		private function fetch_json( $url ) {
+			$response      = \wp_remote_get( $url, array( 'timeout' => 10 ) );
+			$response_code = \wp_remote_retrieve_response_code( $response );
+			$result        = false;
+
+			if ( is_array( $response ) && is_int( $response_code ) && $response_code < 400 ) {
+				$result = json_decode( wp_remote_retrieve_body( $response ) );
```

Comment:
Consider prefixing `wp_remote_retrieve_body()` with a backslash for consistency with other WP functions